internal/controller/api/v1beta1/agent_types.go

```diff
@@ -225,6 +223,12 @@ type DebugInfo struct {
 	// EventsURL specifies an HTTP/S URL that contains events which occured during the cluster installation process
 	// +optional
 	EventsURL string `json:"eventsURL,omitempty"`
+	// +optional
+	// Current state of the Agent
+	State string `json:"state,omitempty"`
```

Comment:
I think the previous place under `AgentStruct` is more proper.
I find that `DebugInfo` should be used only when something goes wrong or an experienced user is looking into more "advanced" fields, and state and state info should be basic fields.

Reply:
@avishayt @filanov WDYT ?

Reply:
I remember that @mhrivnak was against it

Reply:
The information in State should already be represented in the Conditions.  This is providing the information in the format we're used to for help in debugging.